common/beerocks/bwl/shared/nl80211_client_impl.cpp

```diff
@@ -335,6 +335,10 @@ bool nl80211_client_impl::get_radio_info(const std::string &interface_name, radi
                         channel.is_dfs = true;
                     }
 
+                    if (tb_freq[NL80211_FREQUENCY_ATTR_NO_IR]) {
```

Comment:
At some point in time NL80211_FREQUENCY_ATTR_NO_IBSS and NL80211_FREQUENCY_ATTR_PASSIVE_SCAN were replaced by NL80211_FREQUENCY_ATTR_NO_IR. 

Also NL80211_FREQUENCY_ATTR_NO_IBSS was renamed to __NL80211_FREQUENCY_ATTR_NO_IBSS.

From then on, both NL80211_FREQUENCY_ATTR_NO_IR and __NL80211_FREQUENCY_ATTR_NO_IBSS
are set if NO_IR. 

However, in old kernels __NL80211_FREQUENCY_ATTR_NO_IBSS will be present but NL80211_FREQUENCY_ATTR_NO_IR will not.

You should probably do something similar to what the `iw` command line tool is doing:
```c
	/* If both flags are set assume an new kernel */
	if (tb_freq[NL80211_FREQUENCY_ATTR_NO_IR] && tb_freq[__NL80211_FREQUENCY_ATTR_NO_IBSS]) {
		print_flag("no IR", &open);
	} else if (tb_freq[NL80211_FREQUENCY_ATTR_PASSIVE_SCAN]) {
		print_flag("passive scan", &open);
	} else if (tb_freq[__NL80211_FREQUENCY_ATTR_NO_IBSS]){
		print_flag("no ibss", &open);
	}
```

Reply:
Changed the if statement and added a comment to describe it

Reply:
In old kernels __NL80211_FREQUENCY_ATTR_NO_IBSS will be present and NL80211_FREQUENCY_ATTR_NO_IR will not. 

We should decide on how these kernels should be treated. Shall we ignore the channel? Shall we set no_ir to true? Shall we add a no_ibss flag?

Reply:
I guess we should then make it a 4-state enum like the iw tool does: NO_IR, PASSIVE_SCAN, NO_IBSS or nothing at all.

Or, we could make a very simplifying assumption, and say:
- if RADAR is set, we mark it as a DFS channel;
- else if NO_IBSS is set, we ignore the channel.
That more or less corresponds to what hostapd does.

By the way, please note:
```
#define NL80211_FREQUENCY_ATTR_PASSIVE_SCAN	NL80211_FREQUENCY_ATTR_NO_IR
#define NL80211_FREQUENCY_ATTR_NO_IBSS		NL80211_FREQUENCY_ATTR_NO_IR
```
so actually the passive scan case is NO_IR without NO_IBSS. So I assume the split between NO_IR and NO_IBSS was made to support the passive scan case.

Reply:
> if NO_IBSS is set, we ignore the channel.

This is what we had before this PR. The problem is that we have some devices that set the NO_IBSS attribute and others that don't when AFAIK they should behave similarly.

Reply:
Added a filtering check before the end of the iteration
if the channel is not a DFS channel (denoted by the `is_dfs` flag) but still has the NO_IR or NO_IBSS attributes (the `no_ir` is set to `true`) then the channel will be ignored